x-pack/plugins/drilldowns/public/components/flyout_drilldown_wizard/flyout_drilldown_wizard.tsx

```diff
@@ -41,6 +41,72 @@ export interface FlyoutDrilldownWizardProps<CurrentActionConfig extends object =
   actionFactoryContext?: object;
 }
 
+function useWizardConfigState(
+  initialDrilldownWizardConfig?: DrilldownWizardConfig
+): [
+  DrilldownWizardConfig,
+  {
+    setName: (name: string) => void;
+    setActionConfig: (actionConfig: object) => void;
+    setActionFactory: (actionFactory?: ActionFactory) => void;
+  }
+] {
+  const [wizardConfig, setWizardConfig] = useState<DrilldownWizardConfig>(
+    () =>
+      initialDrilldownWizardConfig ?? {
+        name: '',
+      }
+  );
+  const [actionConfigCache, setActionConfigCache] = useState<Record<string, object>>(
```

Comment:
What do you think about having all configs in `DrilldownWizardConfig` instead of introducing `actionConfigCache`?

```ts
export interface DrilldownWizardConfig {
  name: string;
  actionFactory?: ActionFactory;
  configs?: Record<string, object>;
}
```

Reply:
I think I like the current version more, because we leave api of this component simpler, but still have this nice UX touch of not losing changes when switching between factories. 

I wouldn't do it unless we really want to preserve those internal states longer then component lifecycle.